jujugui/static/gui/src/app/components/charmbrowser/test-charmbrowser.js

```diff
@@ -43,14 +43,12 @@ describe('Charmbrowser', function() {
         }
       }};
     var changeState = sinon.stub();
-    var addService = sinon.stub();
     var charmstoreSearch = sinon.stub();
     var output = jsTestUtils.shallowRender(
       <juju.components.Charmbrowser
         appState={appState}
         changeState={changeState}
-        charmstoreSearch={charmstoreSearch}
-        addService={addService} />);
```

Comment:
It's more that `addService` is not being used in this particular test. We really only need to mockup (and eventually assert on) the services that are being used by that particular test. For example, the test to make sure the entity details are loaded properly doesn't need to mockup charmstoreSearch, because search isn't used. It just ends up being an undefined prop, which is OK because it never gets referenced in the entity details branch of code.

That said, we also need to talk about `addService`. I took a look at the underlying code in utils and it's ugly. The full function call is `utils.addService(app, charmId)`, so any component calling it would need app passed in. I don't think we want to pass the entire application down through each component. We need an lightweight (or at leas lighter) implementation of `addService`.